lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -293,45 +330,71 @@ def commit(action, post, authorization = nil)
 
         succeeded = success_from(action, response)
 
-        response(action, succeeded, response)
+        response(action, succeeded, response, source_id)
       end
 
-      def response(action, succeeded, response)
+      def response(action, succeeded, response, source_id = nil)
         successful_response = succeeded && action == :purchase || action == :authorize
         avs_result = successful_response ? avs_result(response) : nil
         cvv_result = successful_response ? cvv_result(response) : nil
 
+        if action == :unstore
+          message_text = "The instrument #{source_id} has been deleted/unstored"
```

Comment:
💬 Comment:

Are you using this particular text for some reason, maybe the gateway just responds with 200 and that's all ?

on side effect for consideration => the merchant is going the get this response and going to think that is the actual message from the gateway, in case of some contact con the gateway support this could lead to misunderstandings.

Reply:
I agree with @Heavyblade here. I actually don't think we need this conditional block. If no value is returned from the gateway, the default value should just be `nil` when we put it in the response. I think the current `message_from` and `authorization_from` helper methods should do that.